Don't force use of the helper for account verification with NIS+; the
previous code already works robustly for any non-threaded caller, and
will fall back to use of the helper anyway.

Authors: Steve Langasek <steve.langasek@canonical.com>

Upstream status: to be discussed

Index: pam.deb/modules/pam_unix/passverify.c
===================================================================
--- pam.deb.orig/modules/pam_unix/passverify.c
+++ pam.deb/modules/pam_unix/passverify.c
@@ -166,7 +166,6 @@
 	if (*pwd != NULL) {
 		if (strcmp((*pwd)->pw_passwd, "*NP*") == 0)
 		{ /* NIS+ */
-#ifdef HELPER_COMPILE
 			uid_t save_euid, save_uid;
 
 			save_euid = geteuid();
@@ -194,10 +193,6 @@
 
 			if (*spwdent == NULL || (*spwdent)->sp_pwdp == NULL)
 				return PAM_AUTHINFO_UNAVAIL;
-#else
-			/* we must run helper for NIS+ passwords */
-			return PAM_UNIX_RUN_HELPER;
-#endif
 		} else if (is_pwd_shadowed(*pwd)) {
 			/*
 			 * ...and shadow password file entry for this user,
